test/pylib/scylla_cluster.py

```diff
@@ -219,6 +221,18 @@ async def install(self) -> None:
 
         self.log_file = self.log_filename.open("wb")
 
+    def get_config(self) -> dict[str, object]:
+        """Return the contents of conf/scylla.yaml as a dict."""
+        return self.config
+
+    def update_config(self, key: str, value: object) -> None:
+        """Update conf/scylla.yaml by setting `value` under `key`.
+           If we're running, reload the config with a SIGHUP."""
+        self.config[key] = value
+        self._write_config_file()
+        if self.cmd:
+            self.cmd.send_signal(signal.SIGHUP)
```

Comment:
Is there a way to wait here for the configuration to really be reloaded?
Also, some options are not live-updatable. I wonder if we won't need to also had an option or a function to restart the server - not just SIGHUP it.

Reply:
> Is there a way to wait here for the configuration to really be reloaded?

We could watch the logs... but for my intended use case I want to watch the logs from the test anyway looking for Raft upgrade logs which should appear after the config is reloaded. So while this feature may be useful for some use cases not everyone can rely on it. If a use case appears that needs to wait for the reload to finish we can extend this function.

>  Also, some options are not live-updatable. I wonder if we won't need to also had an option or a function to restart the server - not just SIGHUP it.

There's already a `Manager` API for restarting servers.